File: License Key Formatting.cpp
class Solution
{
   public:
    string ReFormatString(string s, int k){
    	int n = s.size();
    	int cnt=0;
    	string ans="";
    	for(int i=n-1;i>=0;i--){
    	   if(s[i] == '-'){
    	       if(cnt== k){
    	           cnt=0;
    	           ans+= '-';
    	       }
    	   }else{
    	       if(cnt == k){
    	           ans+= '-';
    	           cnt=0;
    	       }
    	       cnt++;
    	       if(s[i] <= 'z' && s[i] >= 'a'){
    	           ans += char('A' + (s[i]-'a'));
    	       }else{
    	           ans+= s[i];
    	       }
    	   }
    	}
    	if(ans.back() == '-'){
    	    ans.pop_back();
    	}
    	reverse(ans.begin(), ans.end());
    	return ans;
    }
};

File: License Key Formatting.java
class Solution{
    static String ReFormatString(String S, int K){
        StringBuilder ans = new StringBuilder();
        int cnt = 0;
        for(int i = S.length() - 1; i >= 0; i--) {
                char ch = S.charAt(i);
                if(ch != '-' && cnt < K){
                    ans.append(Character.toUpperCase(ch));
                    cnt++;
                }
                if(cnt==K){
                    cnt = 0;
                    ans.append('-');
                }
        }
        ans.reverse();
        if(ans.length() > 0 && ans.charAt(0)=='-') ans.deleteCharAt(0);
        return ans.toString();
        
    }
}
